### ISSUE TITLE
Multiple transactions in single block errors with `MerkleWitness not valid for StateTransition`

### ISSUE THREAD

**Issue by teddyjfpender** [proto-kit/framework#51]
**Describe the bug**
When sending many transactions to an app-chain & then producing a block the test errors with:
```bash
test/Balances.test.ts:
Block Production time: 1696 milliseconds
111 |         if (response.taskId !== undefined) {
112 |             const resolveFunction = this.resultsPending[response.taskId];
113 |             if (!this.erroredOut) {
114 |                 if (response.status === "error") {
115 |                     this.erroredOut = true;
116 |                     this.errorFunction?.(new Error(`Error in worker: ${response.payload}`));
                                            ^
error: Error in worker: MerkleWitness not valid for StateTransition (0, type normal)
Bool.assertTrue(): false != true
```

I came across this error in both `bun` and `npm`.

**To Reproduce**
1. Go to [bun bug repository](https://github.com/teddyjfpender/proto-kit-bun-project) or the [npm bug repository](https://github.com/teddyjfpender/proto-kit-npm-project)
2. Run the tests (`bun install && bun test` or `npm test`)
3. See error

**Expected behavior**
When sending multiple transactions to the app-chain it should produce a block and not error.

**Additional context**

**Comment by rpanic** [comment]
Thanks for the report and the easy reproduction example!

**Comment by rpanic** [comment]
Fixed in #54
